### DISCUSSION TITLE
Not sending form data parameter for UploadFile field throws Internal Error -- How can I handle/fix this?

### DISCUSSION THREAD

**Discussion by dhruv-ahuja** [fastapi/fastapi#9292] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, UploadFile
import uvicorn 

app = FastAPI()


@app.post("/test")
def test_file_upload(data: UploadFile):
    """testing file upload error happening with FastAPI"""

    return {"data": data}


if __name__ == "__main__":
    uvicorn.run("main:app", host="0.0.0.0", port=8000, reload=True)
```

### Description

Running the above while having the `data` parameter unselected in Insomnia is giving the following error in console:
`Did not find CR at end of boundary (21)`

Can someone please guide me towards a solution or a workaround for this? I have been unable to replicate this issue Postman for Mac but with Insomnia and Postman on Windows I've been able to do so.

Insomnia auto-attaches the following header: `"Content-Type": "multipart/form-data"`. 

The following is the error response by FastAPI:
```json
{
	"detail": "There was an error parsing the body"
}
```

The positive output response is like so:
```json
{
	"data": {
		"filename": "sample.csv",
		"file": {},
		"size": 0,
		"headers": {
			"content-disposition": "form-data; name=\"data\"; filename=\"sample.csv\"",
			"content-type": "text/csv"
		}
	}
}
```

Any help would be much appreciated

### Operating System

macOS

### Operating System Details

Mac OS 12.5 Monterey
Macbook Pro 13 Inch M2 2022 machine

### FastAPI Version

0.95.0

### Python Version

3.8.10

### Additional Context
insomnia client image
<img width="900" alt="image" src="https://user-images.githubusercontent.com/83733638/226817998-bd07c9ef-4ea8-43f6-80df-7b43fa59b16a.png">

console image:
<img width="900" alt="image" src="https://user-images.githubusercontent.com/83733638/226817960-99ad2828-12e0-4e57-ba8d-e45951daf450.png">

**Comment by Kludex** [comment]
Maybe related to https://github.com/encode/starlette/issues/1059 ?

**Comment by goophps** [comment]
I met the same problem, how to solve it?
